kore/src/Kore/Repl.hs

```diff
@@ -61,12 +62,13 @@ import           Kore.Unification.Procedure
 -- that would otherwise be required in the proof and allows for step-by-step
 -- execution of proofs. Currently works via stdin/stdout interaction.
 runRepl
-    :: forall claim m
+    :: forall claim axiom m
     .  MonadSimplify m
     => MonadIO m
     => MonadCatch m
     => Claim claim
-    => [Axiom]
+    => axiom ~ Rule claim
```

Comment:
With `axiom ~ Rule claim`, can we remove the second parameters of `ReplState` and `Config`?

Reply:
This would probably mean a lot of changes in the `repl`. Maybe it would be best if I do this in a follow-up PR, so I don't block the merging of the other PRs?